x-pack/plugins/security_solution/public/management/pages/policy/view/policy_list.tsx

```diff
@@ -5,16 +5,199 @@
  * 2.0.
  */
 
-import React, { memo } from 'react';
+import React, { memo, useCallback, useMemo } from 'react';
+import {
+  EuiBasicTable,
+  EuiText,
+  EuiHorizontalRule,
+  EuiAvatar,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiToolTip,
+  CriteriaWithPagination,
+} from '@elastic/eui';
+import { useQuery } from 'react-query';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
 import { AdministrationListPage } from '../../../components/administration_list_page';
+import { useHttp } from '../../../../common/lib/kibana/hooks';
+import { sendGetEndpointSpecificPackagePolicies } from '../../../services/policies/policies';
+import { FormattedDate } from '../../../../common/components/formatted_date';
+import { TextValueDisplay } from '../../../components/artifact_entry_card/components/text_value_display';
+import { EndpointPolicyLink } from '../../endpoint_hosts/view/components/endpoint_policy_link';
+import { PolicyData } from '../../../../../common/endpoint/types';
+import { useUrlPagination } from '../../../components/hooks/use_url_pagination';
 
+const MAX_PAGINATED_ITEM = 9999;
 export const PolicyList = memo(() => {
+  const http = useHttp();
+  const { pagination, pageSizeOptions, setPagination } = useUrlPagination();
+  const result = useQuery(['policyList'], () => {
```

Comment:
Can you move this to a separated file within a hook? There is already one in https://github.com/elastic/kibana/blob/b6753241ed424d236c0ddfcc4f94437c188a7f12/x-pack/plugins/security_solution/public/management/services/policies/hooks.ts, maybe you can reuse it doing a bit of refactor